### PR TITLE
Validator

### LINKED ISSUE #2921: Validate `.data[].secretKey` duplicates
**Describe the bug**
I have same `secretKey` `SERVICE_NAME` but point to diferenet parameters in AWS Parameter Store. I see that operator take the last one in YAML and inject into the secret. So in my example into `test` secret I will go `SERVICE_NAME` from ` /dev/project_env_vars` Parameter instead of `/dev/shared_env_vars` Parameter. 

**To Reproduce**
I have such Externalsecret
```yaml
apiVersion: external-secrets.io/v1alpha1
kind: ExternalSecret
metadata:
  name: test
  namespace: default
spec:
  secretStoreRef:
    name: ssm-secret-store-global
    kind: ClusterSecretStore
  refreshInterval: "3m"
  target:
    name: test-env
    creationPolicy: 'Owner'
  data:
    - secretKey: SERVICE_NAME
      remoteRef:
        key: /dev/shared_env_vars
        property: TEST_URL
    - secretKey: KAFKA_BROKER_0
      remoteRef:
        key: /dev/shared_env_vars
        property: KAFKA_BROKER_0
    - secretKey: KAFKA_BROKER_1
      remoteRef:
        key: /dev/shared_env_vars
        property: KAFKA_BROKER_1
    - secretKey: KAFKA_BROKER_2
      remoteRef:
        key: /dev/shared_env_vars
        property: KAFKA_BROKER_2
    - secretKey: SERVICE_NAME
      remoteRef:
        key: /dev/project_env_vars
        property: SERVICE_NAME
```

**Expected behavior**
Error. I want to see some kind of duplication error. This is not transperent behaviour because you can inject wrong parameter.

Operator version: **0.9.3**

### PR DESCRIPTION
**Pull request by Mehrbod2002**
## Problem Statement

Fix duplicates keys .

## Related Issue

Fixes #2921 and related to #2929

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by Mehrbod2002** [comment]
@shuheiktgw

**Comment by shuheiktgw** [comment]
Sorry, I'm well aware of this pull request, but I've been sick for the last couple of days and am still recovering from it. I'll take a close look when I'm back!

File: apis/externalsecrets/v1beta1/externalsecret_validator_test.go [hunk 1]
```diff
@@ -122,6 +122,21 @@ either data or dataFrom should be specified`,
 				},
 			},
 		},
+		{
```
Comment (Skarlso) [inline apis/externalsecrets/v1beta1/externalsecret_validator_test.go:125]: Add one when there are mutliple matching secrets and some that aren't duplicates please :)

Comment (Mehrbod2002) [reply apis/externalsecrets/v1beta1/externalsecret_validator_test.go:125]: Added :)

Comment (Skarlso) [reply apis/externalsecrets/v1beta1/externalsecret_validator_test.go:125]: ah, meant when there are multiple duplicates, like:

```go

					Data: []ExternalSecretData{
						{SecretKey: "SERVICE_NAME"},
						{SecretKey: "SERVICE_NAME"},
						{SecretKey: "SERVICE_NAME-2"},
						{SecretKey: "SERVICE_NAME-2"},
						{SecretKey: "NOT_DUPLICATE"},
					},
```

This way we can see that it continued and can list multiple problems nicely.

Comment (Mehrbod2002) [reply apis/externalsecrets/v1beta1/externalsecret_validator_test.go:125]: Can you review the error ? 
Test cases return same errors but not match.

Comment (Skarlso) [reply apis/externalsecrets/v1beta1/externalsecret_validator_test.go:125]: It's joined by new line not `,`. So:

```go
expectedErr: "duplicate secretKey found: SERVICE_NAME\nduplicate secretKey found: SERVICE_NAME-2",
```

Also, please fix the now failing sonarcloud error: https://sonarcloud.io/project/issues?resolved=false&pullRequest=3003&id=external-secrets_external-secrets&open=AYzjXXGiDU-oqzasTv1H

Basically, the function is now too complex. :) Do something like:

```go

func validateDuplicateKeys(es *ExternalSecret, errs error) error {
	if es.Spec.Target.DeletionPolicy == DeletionPolicyRetain {
		seenKeys := make(map[string]struct{})
		for _, data := range es.Spec.Data {
			secretKey := data.SecretKey
			if _, exists := seenKeys[secretKey]; exists {
				errs = errors.Join(errs, fmt.Errorf("duplicate secretKey found: %s", secretKey))
			}
			seenKeys[secretKey] = struct{}{}
		}
	}
	return errs
}

```

And then join if errors is not nil. That should be enough.

Comment (Mehrbod2002) [reply apis/externalsecrets/v1beta1/externalsecret_validator_test.go:125]: Done :) @Skarlso

**Comment by shuheiktgw** [comment]
Thank you, @Skarlso to takes it over 🙇

**Review by Skarlso — APPROVED** [review]

**Pull request merged**